x-pack/platform/plugins/shared/streams/server/lib/streams/create_streams_global_search_result_provider.ts

```diff
@@ -0,0 +1,153 @@
+  term,
+  types,
+  maxResults,
+  storageClient,
+  client,
+}: {
+  term: string;
+  types: string[];
+  maxResults: number;
+  storageClient: StreamsStorageClient;
+  client: IScopedClusterClient;
+}) {
+  // This does NOT included unmanaged Classic streams
+  const searchResponse = await storageClient.search({
+    size: maxResults,
+    track_total_hits: false,
+    query: {
+      bool: {
+        should: [
+          {
+            wildcard: {
+              name: {
+                value: `*${term}*`,
+              },
+            },
+          },
+          {
+            match: {
+              description: term,
+            },
+          },
+        ],
+        minimum_should_match: 1,
+      },
+    },
+  });
+
+  const privileges = await checkAccessBulk({
+    names: searchResponse.hits.hits
+      .filter((hit) => !Streams.GroupStream.Definition.is(hit._source))
+      .map((hit) => hit._source.name),
+    scopedClusterClient: client,
+  });
+
+  const hitsWithAccess = searchResponse.hits.hits.filter((hit) => {
+    if (Streams.GroupStream.Definition.is(hit._source)) return true;
+    return privileges[hit._source.name]?.read === true;
+  });
+
+  const allStreams = types.includes('stream');
+  const classicStreams = types.includes('classic stream');
+  const wiredStreams = types.includes('wired stream');
+  const groupStreams = types.includes('group stream');
+  const noSubStreams = !classicStreams && !wiredStreams && !groupStreams;
+
+  if (allStreams || noSubStreams) {
+    return hitsWithAccess.map((hit) => toGlobalSearchProviderResult(hit._id!, hit._source, term));
+  }
+
+  const relevantStreams = hitsWithAccess.filter(({ _source }) => {
+    return (
+      (classicStreams && Streams.ClassicStream.Definition.is(_source)) ||
+      (wiredStreams && Streams.WiredStream.Definition.is(_source)) ||
+      (groupStreams && Streams.GroupStream.Definition.is(_source))
+    );
+  });
+
+  return relevantStreams.map((hit) => toGlobalSearchProviderResult(hit._id!, hit._source, term));
+}
+
+function toGlobalSearchProviderResult(
+  id: string,
+  definition: Streams.all.Definition,
+  term: string
+): GlobalSearchProviderResult {
+  const type = Streams.ClassicStream.Definition.is(definition)
+    ? 'Classic stream'
+    : Streams.WiredStream.Definition.is(definition)
+    ? 'Wired stream'
+    : Streams.GroupStream.Definition.is(definition)
+    ? 'Group stream'
+    : 'Stream';
+
+  return {
+    id,
+    score: calculateTermScore(definition.name.toLowerCase(), term.toLowerCase()),
+    title: definition.name,
+    type,
+    url: `/app/streams/${definition.name}/management/lifecycle`,
+    icon: 'logsApp',
+  };
+}
+
+function calculateTermScore(name: string, searchTerm: string) {
+  if (!searchTerm) {
+    return 80;
+  } else if (name === searchTerm) {
+    return 100;
+  } else if (name.startsWith(searchTerm)) {
+    return 90;
+  } else if (name.includes(searchTerm)) {
+    return 75;
+  }
+
+  return 50;
+}
```

Comment:
Do you think would be beneficial a simple case normalization for the search? Just thinking it would be great to anyway get a high-rank result if the user types `LOGS.CHILD` instead of `logs.child`, that would be anyway a good match.
```ts
const normalizedName = name.toLowerCase();
const normalizedSearchTerm = searchTerm.toLowerCase();
// replace everywhere
```

Reply:
I do this on line 133 just to avoid having the extra variable with a name here :D